packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx

```diff
@@ -170,19 +190,21 @@ export function ThirdwebWalletProvider(
       }
 
       // save to storage
+
       const walletInfo: LastConnectedWalletInfo = {
-        walletId: wallet.walletId,
+        walletId: walletConfig.id,
         connectParams,
       };
 
       // if personal wallet exists, we need to replace the connectParams.personalWallet to a stringifiable version
       const personalWallet = wallet.getPersonalWallet() as AbstractClientWallet;
+      const personalWalletConfig = walletInstanceToConfig.get(personalWallet);
```

Comment:
No - it will not. all the wallets must be connected/used by using either `useConnect` or `useWalletInstance` + `useSetConnectedWallet` + `useSetConnectionStatus`